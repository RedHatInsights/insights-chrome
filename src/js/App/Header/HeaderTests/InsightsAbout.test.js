/* eslint-disable camelcase */
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ConnectedInsightsAbout, { InsightsAbout, Copyright } from '../InsightsAbout';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

describe('InsightsAbout', () => {
    beforeEach(() => {
        fetch.resetMocks();
        fetch.mockResponse(JSON.stringify({ app_name: 'chrome', src_hash: 'f3d0275e021cfa46982fd9a4376bda7a5ee9e079',
            src_tag: '', src_branch: 'prod-stable', travis: { event_type: 'push' },
            build_branch: 'prod-stable', build_hash: 'b5b92cf36f81fec1c9abc302a365bff111b9f5b0', build_id: '7'  }));
    });
    let globalNavData = require('../../../../../testdata/globalNav.json');

    const initialProps = {
        activeApp: 'someApp',
        appId: 'someID',
        dispatch: jest.fn(),
        onClose: jest.fn(),
        user: 'someUser',
        globalNav: [globalNavData]
    };
    it('should render correctly with modal closed', ()=>{
        let props = {
            ...initialProps,
            isModalOpen: false
        };
        shallow(<InsightsAbout { ...props }/>);
    });
    it('should render correctly with modal open', ()=>{
        let props = {
            ...initialProps,
            isModalOpen: true
        };
        shallow(<InsightsAbout { ...props }/>);
    });
});

describe('ConnectedInsightsAbout', () => {
    let initialState;
    let mockStore;
    let globalNavData = require('../../../../../testdata/globalNav.json');
    beforeEach(() => {
        mockStore = configureStore();
        initialState = {
            chrome: {
                user: {
                    identity: {
                        user: {}
                    }
                },
                appId: 'test',
                globalNav: [globalNavData],
                activeApp: 'test'
            }
        };
    });

    it('should render correctly with just username', () => {
        const store = mockStore(initialState);
        const wrapper = mount(
            <Provider store={store}>
                <ConnectedInsightsAbout />
            </Provider>);
        //wrapper.setState({ isOpen: true });
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe('Copyright', () => {
    it('should render', () => {
        const wrapper = shallow(<Copyright/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
