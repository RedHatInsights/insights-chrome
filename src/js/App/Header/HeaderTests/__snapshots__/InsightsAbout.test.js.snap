// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedInsightsAbout should render correctly with just username 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(InsightsAbout)>
    <InsightsAbout
      activeApp="test"
      appId="test"
      dispatch={[Function]}
      globalNav={
        Array [
          Object {
            "ansible": Object {
              "id": "ansible",
              "routes": Array [],
              "title": "Red Hat Ansible Automation Platform",
            },
            "docs": Object {
              "id": "docs",
              "routes": Array [],
              "title": "Documentation",
            },
            "insights": Object {
              "id": "insights",
              "routes": Array [],
              "title": "Red Hat Insights",
            },
            "openshift": Object {
              "id": "openshift",
              "routes": Array [],
              "title": "Red Hat Openshift Cluster Manager",
            },
            "rhel": Object {
              "id": "rhel",
              "routes": Array [],
              "title": "Cloud Management Services",
            },
            "staging": Object {
              "id": "staging",
              "routes": Array [],
              "title": "Staging Bundle",
            },
          },
        ]
      }
      user={Object {}}
    >
      <AboutModal
        appendTo={null}
        backgroundImageSrc=""
        brandImageAlt="Red Hat Logo"
        brandImageSrc="test-file-stub"
        className="ins-c-about-modal"
        isOpen={false}
        noAboutModalBoxContentContainer={false}
        onClose={[Function]}
        productName=""
        trademark={<Copyright />}
      >
        <Portal
          containerInfo={<div />}
        >
          <AboutModalContainer
            appendTo={null}
            ariaDescribedById="pf-about-modal-content-0"
            ariaLabelledbyId="pf-about-modal-title-0"
            backgroundImageSrc=""
            brandImageAlt="Red Hat Logo"
            brandImageSrc="test-file-stub"
            className="ins-c-about-modal"
            isOpen={false}
            noAboutModalBoxContentContainer={false}
            onClose={[Function]}
            productName=""
            trademark={<Copyright />}
          />
        </Portal>
      </AboutModal>
    </InsightsAbout>
  </Connect(InsightsAbout)>
</Provider>
`;

exports[`Copyright should render 1`] = `
<div
  className="ins-c-footer__traditional-nav pf-l-flex pf-m-column pf-m-row-on-lg pf-m-flex-1-on-lg ins-c-page__about--modal-footer"
>
  <p
    className="copyright pf-m-spacer-xl-on-lg"
  >
    Copyright Â© 2019 Red Hat, Inc.
  </p>
  <nav>
    <ul
      className="pf-l-flex pf-m-column pf-m-row-on-md"
    >
      <li>
        <a
          className="nav-link"
          href="https://www.redhat.com/en/about/privacy-policy"
          rel="noopener noreferrer"
          target="_blank"
        >
          Privacy Policy
        </a>
      </li>
      <li>
        <a
          className="nav-link"
          href="https://access.redhat.com/help/terms/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Terms of Use
        </a>
      </li>
      <li>
        <a
          className="nav-link"
          href="https://www.redhat.com/en/about/all-policies-guidelines"
          rel="noopener noreferrer"
          target="_blank"
        >
          All Policies and Guidelines
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
